bugbug/models/duplicate.py

```diff
@@ -28,6 +30,28 @@ def fit(self, X, y):
         self._le.fit(y)
 
 
+class BypassFeatures(TransformerMixin, BaseEstimator):
```

Comment:
Why was this needed?

Reply:
Right now our pipeline looks like:
```
self.extraction_pipeline = Pipeline(
            [
                (
                    "bug_extractor",
                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
                ),
                (
                    "union",
                    ColumnTransformer(
                        [
                            ("text", self.text_vectorizer(), "text"),
                        ]
                    ),
                ),
            ]
        )
```

BugFeatures returns a dictionary {"text":... , "couple_data"..}, but the `ColumnTransformer` in the next step just takes the "text" data from the dict and ignores everything else. Hence, we end up loosing the "couple_data" data, which has all the bug couple features. I confirmed this by completely removing all bug couple data, but I still get same dimensional features as obtained by using them (implying that they weren't being used at all).

One way to circumvent this is by introducing the `BypassFeatures` transformer that appends the couple data features to the current tf-idf text vectors, thus expanding our features. Is there a better way to do this? @marco-c